### PR TITLE
add qc plots

### PR DESCRIPTION
**Pull request by klaricch**

### CODE REVIEW

File: manuscript_analyses/qc_plots/na12878_plots.R [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  theme(strip.background = element_blank(),
+        strip.text = element_text(size = 14, colour = "black", face = "bold"),
+        axis.text.x = element_text(size = 10, color = "black", angle = 90, hjust = 1),
+        axis.text.y = element_text(size = 14, color = "black"),
+        axis.title = element_text(size = 18, face = "bold", color = "black")) +
+  labs(x = "Variant",  y = "Heteroplasmy Level")
+vaf_condordance_variant_type
+
+setwd(opt$plot_directory)
+ggsave(vaf_condordance_variant_type, filename = "na12878_vaf.png", dpi = 300, width = 10, height = 6, units = "in")
```
Comment (gtiao) [inline manuscript_analyses/qc_plots/na12878_plots.R:65]: Is there a reason you don't output pdf as well as png, as you did for the duplciate_qc_plots.R script?

**Review by gtiao — CHANGES_REQUESTED** [review]
Logic all seems correct, and the scripts execute without error -- mostly syntax/formatting comments

File: manuscript_analyses/qc_plots/duplicate_qc_plots.R [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+  make_option(c("-d", "--plot_directory"), type = "character", default = NULL,
+              help = "Directory to which plots should be written", metavar = "character"),
+  make_option(c("-t", "--min_het_threshold"), type = "numeric", default = NULL,
+              help = "Minimum heteroplasmy level to define a variant as a PASS heteroplasmic variant, filter out call if below", metavar = "float"))
+
+opt_parser = OptionParser(option_list = option_list)
+opt = parse_args(opt_parser)
+
+
+# TODO: roxygen2 or other to document parameters?
```
[the remark this answers is not in the document]
Comment (klaricch) [reply manuscript_analyses/qc_plots/duplicate_qc_plots.R:24]: don't think it's needed, will remove

File: manuscript_analyses/qc_plots/duplicate_qc_plots.R [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+mtdnaserver <- read.table(opt$mtdnaserver_vl_changes, sep = '\t', quote = "", header = TRUE)
+mutect <- read.table(opt$mutect_vl_changes, sep = '\t', quote = "", header = TRUE)
+setwd(opt$plot_directory)
+
+# Combine mtDNA-Server and Mutect dataframes
+mtdnaserver <- mutate(mtdnaserver, caller = "MtDNA-Server")
+mutect <- mutate(mutect, caller = "Mutect")
+all <- rbind(mtdnaserver, mutect)
+
+# Filter to samples in both Mutect and mtDNA-Server dataframes (needed because several samples in mtDNA-Server failed to run and want to drop those in the comparison)
```
[the remark this answers is not in the document]
Comment (klaricch) [reply manuscript_analyses/qc_plots/duplicate_qc_plots.R:37]: yes, the results from both callers go through the same processing script which keeps the samples in a consistent order

File: manuscript_analyses/qc_plots/duplicate_qc_plots.R [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+                                            fontfamily = "", fontface = "bold", colour = "black", size = 14,
+                                            angle = 90, lineheight = 0.9, alpha = 1)
+
+combined_plot
+
+ggsave(combined_plot, filename = "scatter_dup_snp.png", dpi = 300, width = 10, height = 6, units = "in")
+ggsave(combined_plot, filename = "scatter_dup_snp.pdf", dpi = 300, width = 10, height = 6, units = "in")
+
+#####################################################
+# Count number of homoplasmic SNPs
```
[the remark this answers is not in the document]
Comment (klaricch) [reply manuscript_analyses/qc_plots/duplicate_qc_plots.R:160]: we only report these numbers for Mutect in the paper

File: manuscript_analyses/qc_plots/mixins_plots.R [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  theme(strip.background = element_blank(),
+        strip.text.y = element_text(size = 12, colour = "black"),
+        axis.text.x = element_text(size = 14, angle = 90, colour = "black"),
+        axis.title = element_text(colour = "black", size = 18, face = "bold"),
+        axis.text = element_text(colour = "black", size = 14)) +
+  labs(x = "Heteroplasmy Level", y = "Count")
+heteroplasmy_hist
+
+setwd(opt$plot_directory)
+ggsave(heteroplasmy_hist, filename = "heteroplasmy_hist_mixin_snps.png", dpi = 300, width = 10, height = 6, units = "in")
```
[the remark this answers is not in the document]
Comment (klaricch) [reply manuscript_analyses/qc_plots/mixins_plots.R:31]: i only output a pdf if it's a figure for the paper, we sometimes use the others just for qc

File: manuscript_analyses/qc_plots/na12878_plots.R (hunk 1 above)
Comment (klaricch) [reply manuscript_analyses/qc_plots/na12878_plots.R:65]: i only output a pdf if it's a figure for the paper, we sometimes use the others just for qc

File: manuscript_analyses/qc_plots/duplicate_qc_plots.R [hunk 6]
```diff
@@ -0,0 +1,350 @@
...
+indel_comparison_plot
+
+ggsave(indel_comparison_plot, filename = "scatter_dup_indel.png", dpi = 300, width = 10, height = 6, units = "in")
+ggsave(indel_comparison_plot, filename = "scatter_dup_indel.pdf", dpi = 300, width = 10, height = 6, units = "in")
+
+
+#####################################################
+# Generate jaccard index table
+#####################################################
+add_jaccard_results <- function(intersect, union, jaccard_table, row_name) {
```
[the remark this answers is not in the document]
Comment (klaricch) [reply manuscript_analyses/qc_plots/duplicate_qc_plots.R:257]: hmm yeah used interchangeably here

File: manuscript_analyses/qc_plots/duplicate_qc_plots.R [hunk 7]
```diff
@@ -0,0 +1,350 @@
...
+s1_hom + s2_hom
+print("Variants called(SNV duplicates Mutect):")
+(s1_called + s2_called)
+print("Fraction homoplasmic (SNV duplicates Mutect):")
+(s1_hom + s2_hom)/(s1_called + s2_called)
+
+#####################################################
+# Plot SNPs and indels together for just Mutect
+#####################################################
+# Do not run this analysis for mtDNA-Server since indel-calling is in beta
```
[the remark this answers is not in the document]
Comment (klaricch) [reply manuscript_analyses/qc_plots/duplicate_qc_plots.R:179]: added to comment, still analyzing indels from mtDNA-Server, but separately from SNPs

File: manuscript_analyses/qc_plots/duplicate_qc_plots.R [hunk 8]
```diff
@@ -0,0 +1,350 @@
...
+  theme(strip.background = element_blank(),
+        plot.title = element_text(hjust = 0.5, size = 20, face = "bold"),
+        strip.text.x = element_text(size = 18, colour = "black"),
+        panel.spacing = unit(1.5, "lines"),
+        axis.title = element_text(colour = "black", size = 18, face = "bold"),
+        axis.text = element_text(colour = "black", size = 12)) +
+  labs(x = "Heteroplasmic Level\nSample 1", y = "Heteroplasmic Level\nSample 2", title = dup_label)
+indel_comparison_plot
+
+ggsave(indel_comparison_plot, filename = "scatter_dup_indel.png", dpi = 300, width = 10, height = 6, units = "in")
```
[the remark this answers is not in the document]
Comment (klaricch) [reply manuscript_analyses/qc_plots/duplicate_qc_plots.R:250]: i think i'll actually remove the contour lines as we switched to using the density plots when needed for the paper

File: manuscript_analyses/qc_plots/duplicate_qc_plots.R [hunk 9]
```diff
@@ -0,0 +1,350 @@
...
+                                        fontfamily = "", fontface = "bold", colour = "black", size = 14,
+                                        angle = 90, lineheight = 0.9, alpha = 1)
+
+combined_plot <- combined_plot + draw_label("Density", x = .035, y = 0.90, hjust = .5, vjust = .5,
+                                            fontfamily = "", fontface = "bold", colour = "black", size = 14,
+                                            angle = 90, lineheight = 0.9, alpha = 1)
+
+combined_plot
+
+ggsave(combined_plot, filename = "scatter_dup_snp.png", dpi = 300, width = 10, height = 6, units = "in")
```
[the remark this answers is not in the document]
Comment (klaricch) [reply manuscript_analyses/qc_plots/duplicate_qc_plots.R:155]: i think i'll actually remove the contour lines as we switched to using the density plots when needed for the paper

File: manuscript_analyses/qc_plots/mixins_plots.R (hunk 5 above)
[the remark this answers is not in the document]
Comment (klaricch) [reply manuscript_analyses/qc_plots/mixins_plots.R:31]: I've relabelled them for plotting, but usually leave them as is to be consistent with the output from dsp

**Review by gtiao — CHANGES_REQUESTED** [review]
Looks great! A few tiny nitpicks, and you're ready to merge. Making those plotting functions that you re-use really cleans things up nicely!

**Review by gtiao — APPROVED** [review]

**Pull request merged**